src/audio/code_8019AF00.c

```diff
@@ -4870,37 +4877,457 @@ void Audio_SplitBgmChannels(s8 volumeSplit) {
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A1F88.s")
+void Audio_PlayObjSoundFanfare(Vec3f* pos, s8 seqId) {
+    s32 requestFanfare = false;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A1FB4.s")
+    if (sObjSoundFanfareSeqId == NA_BGM_GENERAL_SFX) {
+        // No spatial fanfare is currently playing
+        requestFanfare = true;
+    } else if (sqrtf(SQ(pos->z) + ((SQ(pos->x) * 0.25f) + (SQ(pos->y) / 6.0f))) <
+               sqrtf(SQ(sObjSoundFanfarePos.z) +
+                     ((SQ(sObjSoundFanfarePos.x) * 0.25f) + (SQ(sObjSoundFanfarePos.y) / 6.0f)))) {
+        // The spatial fanfare requested is closer than the spatial fanfare currently playing
+        requestFanfare = true;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A2090.s")
+    if (requestFanfare) {
+        sObjSoundFanfarePos.x = pos->x;
+        sObjSoundFanfarePos.y = pos->y;
+        sObjSoundFanfarePos.z = pos->z;
+        sObjSoundFanfareSeqId = seqId;
+        sObjSoundFanfareRequested = true;
+    }
+}
+
+void Audio_UpdateObjSoundFanfare(void) {
+    if (sObjSoundFanfareRequested && (sAudioPauseState == AUDIO_PAUSE_STATE_CLOSED)) {
+        if (sObjSoundFanfareSeqId != NA_BGM_GENERAL_SFX) {
+            Audio_StartObjSoundFanfare(SEQ_PLAYER_FANFARE, &sObjSoundFanfarePos, sObjSoundFanfareSeqId, 0);
+
+            if (AudioSeq_GetActiveSeqId(SEQ_PLAYER_FANFARE) == NA_BGM_DISABLED) {
+                func_801A3038();
+            }
+
+            if ((AudioSeq_GetActiveSeqId(SEQ_PLAYER_BGM_MAIN) != NA_BGM_DISABLED) &&
+                (AudioSeq_GetActiveSeqId(SEQ_PLAYER_AMBIENCE) == NA_BGM_DISABLED)) {
+                Audio_PlayAmbience(AMBIENCE_ID_09);
+            }
+
+            sAudioCutsceneFlag = true;
+
+        } else {
+            Audio_StartObjSoundFanfare(SEQ_PLAYER_FANFARE, NULL, sObjSoundFanfareSeqId, 0);
+            if (AudioSeq_GetActiveSeqId(SEQ_PLAYER_BGM_MAIN) != NA_BGM_DISABLED) {
+                SEQCMD_STOP_SEQUENCE(SEQ_PLAYER_AMBIENCE, 0);
+            }
+            sObjSoundFanfareRequested = false;
+            sObjSoundMainBgmSeqId = NA_BGM_GENERAL_SFX;
+            sAudioCutsceneFlag = false;
+        }
+        sObjSoundFanfareSeqId = NA_BGM_GENERAL_SFX;
+    }
+}
+
+// ======== END Z_OBJ_SOUND FUNCTIONS ========
+
+void Audio_StopSequenceAtPos(u8 seqPlayerIndex, u8 volumeFadeTimer) {
+    SEQCMD_STOP_SEQUENCE((u32)seqPlayerIndex, 20);
+    if ((seqPlayerIndex == SEQ_PLAYER_BGM_SUB) && (AudioSeq_GetActiveSeqId(SEQ_PLAYER_BGM_MAIN) != NA_BGM_DISABLED)) {
+        AudioSeq_SetVolumeScale(SEQ_PLAYER_BGM_MAIN, VOL_SCALE_INDEX_BGM_SUB, 0x7F, volumeFadeTimer);
+        Audio_SplitBgmChannels(0);
+    }
+}
+
+void Audio_StartSubBgmAtPos(u8 seqPlayerIndex, Vec3f* pos, u8 seqId, u8 flags, f32 minDist, f32 maxDist, f32 arg6) {
+    f32 dist = sqrtf(SQ(pos->z) + ((SQ(pos->x) * 0.25f) + (SQ(pos->y) / 6.0f)));
```

Comment:
huh, Do we know why it does a weighted dist?

Reply:
This is as projectedPos (which maybe it should be named as that), so z would be in/out of screen, x would be left/right and y would be up/down. So intuitively you would want the source to have less volume change if it moves left/right/up/down vs in/out

Reply:
Ohh, I see, interesting.
A comment explaining it would be very nice to have